src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -792,41 +806,63 @@ CHIP_ERROR DefaultSceneTableImpl::RemoveFabric(FabricIndex fabric_index)
 {
     VerifyOrReturnError(IsInitialized(), CHIP_ERROR_INTERNAL);
 
-    FabricSceneData fabric(fabric_index);
-    SceneIndex idx = 0;
-    CHIP_ERROR err = fabric.Load(mStorage);
-    VerifyOrReturnError(CHIP_NO_ERROR == err || CHIP_ERROR_NOT_FOUND == err, err);
-
-    while (idx < mMaxScenesPerFabric)
+    for (auto endpoint : app::EnabledEndpointsWithServerCluster(chip::app::Clusters::Scenes::Id))
```

Comment:
Just because an endpoint is disabled doesn't mean we should leave its scenes in storage forever.

Also, do we need to worry about dynamic endpoints that change whether scenes are supported?  I'm hoping not...